src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbApplicationReviewRepository.cs

```diff
@@ -48,109 +51,110 @@ public async Task<ApplicationReview> GetAsync(long vacancyReference, Guid candid
         {
             var collection = GetCollection<ApplicationReview>();
 
             var builder = Builders<ApplicationReview>.Filter;
             var filter = builder.Eq(r => r.EmployerAccountId, employerAccountId) &
                          builder.Ne(r => r.IsWithdrawn, true);
 
-            var aggregate = collection.Aggregate()
+            var result = await RetryPolicy.ExecuteAsync(_ =>
+            {
+                var aggregate = collection.Aggregate()
 
-                .Match(filter)
+                    .Match(filter)
 
-                .Group(groupBy => new ApplicationReviewCount.ApplicationReviewsCountGroupKey
-                    {
-                        VacancyReference = groupBy.VacancyReference,
-                        Status = groupBy.Status
-                    },
-                    g =>
-                        new ApplicationReviewCount
-                        {
-                            Id = g.Key,
-                            Count = g.Count()
-                        })
-                .ToListAsync();
+                    .Group(groupBy => new ApplicationReviewCount.ApplicationReviewsCountGroupKey {VacancyReference = groupBy.VacancyReference, Status = groupBy.Status},
+                        g =>
+                            new ApplicationReviewCount {Id = g.Key, Count = g.Count()})
+                    .ToListAsync();
 
-            return aggregate;
+                return aggregate;
+            },
+            new Context(nameof(GetStatusCountsForEmployerAsync)));
+
+            return result;
         }
 
-        public Task<List<ApplicationReviewCount>> GetStatusCountsForProviderAsync(long ukprn)
+        public async Task<List<ApplicationReviewCount>> GetStatusCountsForProviderAsync(long ukprn)
         {
-            // var collection = GetCollection<ApplicationReview>();
+            //var collection = GetCollection<ApplicationReview>();
+
+            //var builder = Builders<ApplicationReview>.Filter;
+            //var filter = builder.Eq(r => r.Ukprn, ukprn) &
+            //             builder.Ne(r => r.IsWithdrawn, true);
 
-            // var builder = Builders<ApplicationReview>.Filter;
-            // var filter = builder.Eq(r => r.Ukprn, ukprn) &
-            //              builder.Ne(r => r.IsWithdrawn, true);
+            //var result = await RetryPolicy.ExecuteAsync(_ =>
+            //{
+            //    var aggregate = collection.Aggregate()
 
-            // var aggregate = collection.Aggregate()
+            //        .Match(filter)
 
-            //     .Match(filter)
+            //        .Group(groupBy => new ApplicationReviewCount.ApplicationReviewsCountGroupKey {VacancyReference = groupBy.VacancyReference, Status = groupBy.Status},
+            //            g =>
+            //                new ApplicationReviewCount {Id = g.Key, Count = g.Count()})
+            //        .ToListAsync();

```

Comment:
Does the commented out code need to be there? Can it be deleted?

Reply:
This will be addressed in my Application Review Count aggregate query PR.